core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/asterixdb/AsterixDBSourceOpExec.scala

```diff
@@ -0,0 +1,331 @@
+    DateTimeFormatter.ISO_LOCAL_DATE_TIME.withZone(ZoneId.from(ZoneOffset.UTC))
+
+  var curQueryString: Option[String] = None
+  var curResultIterator: Option[Iterator[AnyRef]] = None
+
+  /**
+    * A generator of a Texera.Tuple, which converted from a CSV row of fields from AsterixDB
+    * @return Iterator[Tuple]
+    */
+  override def produceTexeraTuple(): Iterator[Tuple] = {
+    new Iterator[Tuple]() {
+      override def hasNext: Boolean = {
+
+        cachedTuple match {
+          // if existing Tuple in cache, means there exist next Tuple.
+          case Some(_) => true
+          case None    =>
+            // cache the next Tuple
+            cachedTuple = Option(next())
+            cachedTuple.isDefined
+        }
+      }
+
+      @throws[RuntimeException]
+      override def next: Tuple = {
+        // if has the next Tuple in cache, return it and clear the cache
+        cachedTuple.foreach(tuple => {
+          cachedTuple = None
+          return tuple
+        })
+
+        // otherwise, send query to fetch for the next Tuple
+        curResultIterator match {
+          case Some(resultSet) =>
+            if (resultSet.hasNext) {
+
+              // manually skip until the offset position in order to adapt to progressive batches
+              curOffset.fold()(offset => {
+                if (offset > 0) {
+                  curOffset = Option(offset - 1)
+                  return next
+                }
+              })
+
+              // construct Texera.Tuple from the next result.
+              val tuple = buildTupleFromRow
+
+              if (tuple == null)
+                return next
+
+              // update the limit in order to adapt to progressive batches
+              curLimit.fold()(limit => {
+                if (limit > 0) {
+                  curLimit = Option(limit - 1)
+                }
+              })
+              tuple
+            } else {
+              // close the current resultSet and query
+              curResultIterator = None
+              curQueryString = None
+              next
+            }
+          case None =>
+            curQueryString = generateSqlQuery
+            curQueryString match {
+              case Some(query) =>
+                curResultIterator = queryAsterixDB(host, port, query)
+                next
+              case None =>
+                curResultIterator = None
+                null
+            }
+        }
+
+      }
+    }
+  }
+
+  /**
+    * close curResultIterator, curQueryString
+    */
+  override def close(): Unit = {
+    curResultIterator = None
+    curQueryString = None
+  }
+
+  /**
+    * add naive support for full text search.
+    * input is either
+    *     ['a','b','c'], {'mode':'any'}
+    * or
+    *     ['a','b','c'], {'mode':'all'}
+    * @param queryBuilder queryBuilder for concatenation
+    * @throws RuntimeException if attribute does not support string based search
+    */
+  @throws[RuntimeException]
```

Comment:
can you add the syntax to opdesc operator description so the user can see the syntax?

Reply:
added